src/webots/gui/WbWrenWindow.cpp

```diff
@@ -269,47 +269,46 @@ void WbWrenWindow::resizeWren(int width, int height) {
   emit resized();
 }
 
-void WbWrenWindow::flipImageBuffer(unsigned char *buffer, int width, int height, int channels) {
-  // flip vertically the image (about 2x faster than QImage::mirrored())
-  const int lineSize = width * channels;
-  const int halfHeight = height / 2;
-  unsigned char *srcPtr = buffer;
-  unsigned char *dstPtr = &buffer[(height - 1) * lineSize];
-  for (int y = 0; y < halfHeight; y++) {
-    for (int x = 0; x < lineSize; x++) {
-      const uchar d = dstPtr[x];
-      dstPtr[x] = srcPtr[x];
-      srcPtr[x] = d;
-    }
-    srcPtr += lineSize;
-    dstPtr -= lineSize;
-  }
+void WbWrenWindow::flipAndScaleDownImageBuffer(const unsigned char *source, unsigned char *destination, int sourceWidth,
+                                               int sourceHeight, int scaleDownFactor) {
+  // flip vertically the image and scale it down (about 3x faster than QImage::mirrored(), QImage::scaled())
+  const int h = sourceHeight / scaleDownFactor;
+  const int w = sourceWidth / scaleDownFactor;
+  const int yFactor = scaleDownFactor * sourceWidth;
+  const uint32_t *src = (const uint32_t *)source;  // assuming a pixel is coded as four bytes (RGBA)
+  uint32_t *dst = (uint32_t *)destination;
+  for (int y = 0; y < h; y++)
+    for (int x = 0; x < w; x++)
+      dst[(h - 1 - y) * w + x] = src[y * yFactor + x * scaleDownFactor];
```

Comment:
This is not respecting our coding style ([CS360](https://github.com/omichel/webots/wiki/CPP-Coding-Style#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets)).

Reply:
I fix also this in my next commit.